Include/pyerrors.h

```diff
@@ -219,8 +219,6 @@ PyAPI_DATA(PyObject *) PyExc_IOError;
 PyAPI_DATA(PyObject *) PyExc_WindowsError;
 #endif
 
-PyAPI_DATA(PyObject *) PyExc_RecursionErrorInst;
```

Comment:
For information, the PyExc_MemoryErrorInst singleton, **also a part of the public API then**, has been removed in 98e2b452975f13b13052aeb9d166ed8a3ef45063 when fixing bpo-5437, and it was removed for the same reasons that this PR is removing the PyExc_RecursionErrorInst singleton.